public/calendar_reboot.js

```diff
@@ -923,17 +923,58 @@ var firebaseCalendarStorage = function(user) {
         
         var userId = self.user.currentUser.uid;
         self.startWork();
-        return self.database.ref('/users/' + userId + '/checkit_allCalendarIds')
-        .once('value')  
-        .then(function(allCalendarIds) {
-            self.endWork();
-            return allCalendarIds.val();
-        })
-        .catch(function() {
-            console.log("allCalendarIds not found");
-            self.endWork();
+        return new Promise(function(resolve, reject) {
+            self.database.ref('/users/' + userId + '/checkit_allCalendarIds')
+            .once('value')  
+            .then(function(allCalendarIds) {
+                self.endWork();
+                if (allCalendarIds.val() === null) {
+                    reject("allCalendarIds not found");
+                }
+                else {
+                    return resolve(allCalendarIds.val());
+                }
+            })
+            .catch(function(err) {
+                self.endWork();
+                console.error(err);
+                return err;
+            })
         })
     };
+    
+    self.save = function(calendarObj) {
+        //save an App object (like a calendar object for example) in storage
+        
+        // Store the calendar state in the firebase database.
+        
```

Comment:
I think it will make things like sharing harder later if every calendar is "part of" a user, but if you're having trouble getting it to work at all like it is now, or you have a better reason to do it, then go for it.